### PR TITLE
upgrade scalafix/scalameta/fastparse/scalapb (and add metaconfig and scala-xml-quote)

### PR DESCRIPTION
**Pull request by SethTisue**
test run: https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1920/consoleFull

### CODE REVIEW

**Comment by SethTisue** [comment]
again: https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1921/consoleFull

**Comment by SethTisue** [comment]
yuck `[scalameta] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.9/project-builds/scalameta-9b55320422005854690113f30ef9922eead3bf9c/scalameta/semantic/jvm/target/scala-2.12/src_managed/main/org/scalameta/semantic/v1/proto/Address.scala:95:
  object creation impossible, since method nestedMessagesCompanions in trait GeneratedMessageCompanion of type => Seq[com.trueaccord.scalapb.GeneratedMessageCompanion[_]] is not defined`

**Comment by SethTisue** [comment]
the whole scalameta-scalafix nexus is so fragile, maybe it's better if we  abandon the big upgrade and stick with the old frozen versions, only tweaked as necessary to be green, rather than refreshed

**Comment by SethTisue** [comment]
hmm, well I couldn't resist at least trying to refresh scalameta & scalafix. I've already succeeded at refreshing scalameta, so it's a question of whether scalafix can be refreshed at the same time. (just ejecting scalafix for now is a possibility)

**Comment by SethTisue** [comment]
omg it works 🎉

**Comment by SethTisue** [comment]
behold, a rabbit! (way down here at the bottom of this long rabbit hole)

**Comment by SethTisue** [comment]
all four upgraded projects remain forked, but from much newer fork points. fyi @xeno-by @lihaoyi @olafurpg @thesamet

**Comment by SethTisue** [comment]
oh and fyi @densh on the addition of scala-xml-quote (using @allanrenucci's fork)

**Comment by densh** [comment]
@SethTisue 👍

File: configs/community.dbuild [hunk 1]
```diff
@@ -403,17 +403,14 @@ build += {
...
 
-  // forked temporarily (September 2017) until this is merged:
-  // https://github.com/lihaoyi/fastparse/pull/166
+  # forked (September 2017) to exempt a failing test file
   ${vars.base} {
     name: "fastparse"
     uri:  ${vars.uris.fastparse-uri}
     extra.projects: [
       "fastparseJVM"   // no Scala.js plz
       "scalaparseJVM"  // dependency of Scala.meta (!)
```
Comment (olafurpg) [inline configs/community.dbuild:412]: No longer, we only depend on fastparseJVM

File: configs/community.dbuild [hunk 2]
```diff
@@ -1007,62 +1004,74 @@ build += {
...
+  // caution, refreshing the fork any further can be problematic;
+  // see Olafur's remarks at
   // https://github.com/scala/community-builds/issues/499#issuecomment-287307613
   // about the instability of both scalameta and scalafix
   ${vars.base} {
     name: "scalafix"
     uri:  ${vars.uris.scalafix-uri}
     extra.exclude: [
       "scalafix-sbt"  // we never build sbt plugins
-      // "These tests depend on various projects (circe, shapeless,...) to
```
Comment (olafurpg) [inline configs/community.dbuild:1052]: 👍 These tests have been removed.

File: configs/community.dbuild [hunk 3]
```diff
@@ -1007,62 +1004,74 @@ build += {
...
-      // it "involves cloning other projects and compiling them"
-      "scalafix-tests"
       // Missing dependency: the library com.github.alexarchambault#case-app
       "cli"
       // requires scalatex-site which requires Scala.js
       "readme"
+      // no Scala.js please
+      "coreJS", "diffJS"
+      // Missing dependency: the library org.scalameta#semanticdb-sbt
+      "testsInputSbt"
```
Comment (olafurpg) [inline configs/community.dbuild:1048]: This is a 2.10 only project.

File: configs/community.dbuild [hunk 4]
```diff
@@ -1007,62 +1004,74 @@ build += {
...
+      // do I *look* like Dotty?
+      "testsOutputDotty"
+      // can't expand macros compiled by previous versions of Scala
+      "testsOutputSbt"
```
Comment (olafurpg) [inline configs/community.dbuild:1052]: This one depends on sbt.

File: configs/community.dbuild [hunk 5]
```diff
@@ -1007,62 +1004,74 @@ build += {
...
+      // Missing dependency: the library org.scalameta#semanticdb-sbt-runtime is not provided
```
Comment (olafurpg) [inline configs/community.dbuild:1053]: This is a small library in https://github.com/scalameta/semanticdb-sbt the `unit` project is where most of the interesting scalafix unit tests exist.

**Review by olafurpg — COMMENTED** [review]
LGTM 👍 Scalafix no longer depends on scalameta/paradise macro annotations so that should simplify a bit w/ regards to the community build.

**Comment by SethTisue** [comment]
@olafurpg thanks for reviewing this. I based 855c3cab02fd81868da491a80edc342aef2161fd on some of your remarks, and opened #608 on the possibility of adding semanticdb-sbt-runtime

**Pull request merged**